xdsl/dialects/riscv.py

```diff
@@ -302,7 +303,7 @@ def __init__(
         comment: str | StringAttr | None = None,
     ):
         if isinstance(immediate, int):
-            immediate = IntegerAttr(immediate, 32)
+            immediate = IntegerAttr(immediate, i32)
```

Comment:
This shouldn't have to be changed, can we look more into it?
I'm confused why this is not inferred to be `IntegerAttr[IntegerType]` anymore.

Reply:
Well, here I would like to understand why this is broken.
Because this currently means that we have a constructor that's unusable, and that's not good at all.